hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtDistributionSetMapper.java

```diff
@@ -117,6 +117,10 @@ static MgmtDistributionSet toResponse(final DistributionSet distributionSet) {
         response.add(linkTo(methodOn(MgmtDistributionSetRestApi.class).getDistributionSet(response.getDsId()))
                 .withSelfRel());
 
+        return response;
+    }
+
+    static void addLinks(final DistributionSet distributionSet, final MgmtDistributionSet response) {
```

Comment:
the method `addLinks`  could return the `response` then you could chain the calls in the actual resources implemenation instead of calling 
```
MgmtDistributionSet response = toResponse(set);
addLinks(set, response);
Response.ok(response);
```

```
MgmtDistributionSet response = addLinks(set, toResponse(set));
Response.ok(response);
```

Reply:
Good point